client/clientmanager/src/main/java/io/mosip/registration/clientmanager/repository/GlobalParamRepository.java

```diff
@@ -3,26 +3,70 @@
 import io.mosip.registration.clientmanager.dao.GlobalParamDao;
 import io.mosip.registration.clientmanager.entity.GlobalParam;
 
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
 import javax.inject.Inject;
 
 public class GlobalParamRepository {
 
-    private GlobalParamDao globalParamDao;
+    private static final String TAG = GlobalParamRepository.class.getSimpleName();
+
+    private static Map<String, String> globalParamList;
+
+    private static GlobalParamDao globalParamDao;
 
     @Inject
     public GlobalParamRepository(GlobalParamDao globalParamDao) {
-        this.globalParamDao = globalParamDao;
+        GlobalParamRepository.globalParamDao = globalParamDao;
     }
 
     public String getGlobalParamValue(String id) {
-        return this.globalParamDao.getGlobalParam(id);
+        return globalParamDao.getGlobalParam(id);
     }
 
     public void saveGlobalParam(String id, String value) {
         GlobalParam globalParam = new GlobalParam(id);
         globalParam.setName(id);
         globalParam.setValue(value);
         globalParam.setStatus(true);
-        this.globalParamDao.insertGlobalParam(globalParam);
+        globalParamDao.insertGlobalParam(globalParam);
+        refreshGlobalParams();
+    }
+
+    public void saveGlobalParams(List<GlobalParam> globalParam) {
+        globalParamDao.insertAll(globalParam);
+        refreshGlobalParams();
+    }
+
+    public List<GlobalParam> getGlobalParams() {
+        return globalParamDao.getGlobalParams();
+    }
+
+    private static void refreshGlobalParams() {
+        globalParamList = new HashMap<>();
```

Comment:
Should we set to new hashmap everytime ?
As its map why not directly update the values in them.

Reply:
done